app/src/pages/Courses.tsx

```diff
@@ -80,36 +124,122 @@ const Courses = () => {
+                        </svg>
+                    </div>
+                  </button>
 
-                {/* TODO: Implement filters */}
+                </div>
+                
+                  <form className="flex flex-col md:flex-row w-3/4 md:w-full max-w-full md:space-x-4 space-y-3 md:space-y-0 justify-end py-6">
+                  <div className="relative min-w-[225px] flex-grow-0">
+                    <input
+                      className="placeholder:text-slate-400 block bg-white w-full border border-slate-300 rounded-md py-2 pr-3 shadow-sm focus:outline-none hover:bg-white focus:border-sky-500 focus:ring-1 sm:text-sm"
+                      type="text"
+                      id="search-term"
+                      placeholder="Buscar curso"
+                      onChange={(event) => setSearchTerm(event.target.value)}
+                    />
+                    <svg
+                      className="absolute right-3 top-1/2 transform -translate-y-1/2 h-5 w-5 text-slate-400"
+                      xmlns="http://www.w3.org/2000/svg"
+                      fill="none"
+                      viewBox="0 0 24 24"
+                      stroke="currentColor"
+                    >
+                      <path
+                        strokeLinecap="round"
+                        strokeLinejoin="round"
+                        strokeWidth="2"
+                        d="M21 21l-4.35-4.35m1.35-5.65a7 7 0 11-14 0 7 7 0 0114 0z"
+                      />
+                    </svg>
+                  </div>
+                  <select
+                    onChange={(event) => setSelectedFilter(event.target.value)}
+                    className="block bg-white min-w-[175px] flex-grow-0 border border-slate-300 rounded-md py-2 pr-3 shadow-sm focus:outline-none hover:bg-white focus:border-sky-500 focus:ring-1 sm:text-sm">
+                    <option value="newest">Últimos incluídos</option>
+                    <option value="oldest">Mais antigos</option>
+                    <option value="popular">Mais populares</option>
+
+                  </select>
+                </form>
+                
               </div>
               <div className="overflow-y-scroll no-scrollbar">
+                
+                {isGridView ? (
                 <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 3xl:grid-cols-4 gap-6 pb-4">
-                  {data.map((course: any, key: number) => (
-                    <CourseListCard course={course} key={key} />
+                {filteredCourses.map((course: any, key: number,) => (
+                    <CourseGridCard course={course} key={key}/>
                   ))}
                 </div>
+                ): (
+                  <table className="w-[100%] leading-normal mx-auto">
+                    <thead>
+                      <tr className="bg-white border-b-4 border-[#166276] text-[#166276] text-left text-base font-base font-['Lato']]">
```

Comment:
The project color `primary` should be used here.  Also, why use `font-['Lato']` instead of `Montserrat`?